cfme/storage/volume.py

```diff
@@ -250,21 +265,35 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for Volume to disappear, continuing')
 
-    def update(self, updates):
+    def update(self, updates, storage_manager=None):
         """Edit cloud volume"""
-        view = navigate_to(self, 'Edit')
-        view.fill({'volume_name': updates.get('name')})
+        if storage_manager:
+            view = navigate_to(storage_manager, 'Volumes')
+            view.entities.get_entity(name=self.name).check()
+            view.toolbar.configuration.item_select('Edit selected Cloud Volume')
+            view = view.browser.create_view(VolumeEditView, additional_context={'object': self})
+        else:
+            view = navigate_to(self, 'Edit')
+        view.fill(updates)
 
         view.save.click()
-
-        view.flash.assert_success_message('Cloud Volume "{}" updated'.format(updates.get('name')))
+        view.flash.assert_success_message('Cloud Volume "{}" updated'.format(
+            updates.get('volume_name')))
         wait_for(lambda: not self.exists, delay=20, timeout=500, fail_func=self.refresh)
+        volume_collection = self.appliance.collections.volumes
+        return volume_collection.instantiate(
```

Comment:
This is unnecessary, if `Updateable` is used correctly.  The object is updated within the context manager in place.

Reply:
Then maybe I used it incorrectly.
I tried it with update method as it was, but it was failing.
Volume name in object wasn't changed so I needed to instantiate it.

Reply:
Also I have to use volume.update because there are multiple possible navigations specified.